pandas/_libs/parsers.pyx

```diff
@@ -1677,7 +1677,7 @@ cdef _try_double(parser_t *parser, int64_t col,
 cdef int _try_double_nogil(parser_t *parser,
                            float64_t (*double_converter)(
                                const char *, char **, char,
-                               char, char, int, int *, int *) nogil,
+                               char, char, int, int *, int *) noexcept nogil,
```

Comment:
Based on https://github.com/pandas-dev/pandas/pull/55650#issuecomment-1776033866, don't we not want noexcept here? It's still not clear to me why we don't though.

Reply:
Good callout - this PR just got us back to 0.29 behavior but we could do better with the error handling. The problem is the body of the function returns 1 on error and doesn't set a Python exception. We at least would need to return -1 and I think set a Python exception in an ideal world